harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -679,16 +562,33 @@ def _load(self) -> None:
                 break
 
         self.context.model.load_state_dict(checkpoint["model_state_dict"])
-        self.context.optimizer.load_state_dict(checkpoint["optimizer_state_dict"])
-        if self.context.lr_scheduler is not None:
-            self.context.lr_scheduler.load_state_dict(checkpoint.get("lr_scheduler"))
+
+        # TODO(DET-3262): remove this backward compatibility of supporting the old schema
+        #     of checkpointing the only optimizer.
```

Comment:
We should plan on supporting loading old models for... well for quite a while.  We have customers with old checkpoints in production, and they would not appreciate if we broke that for them.  The precise answer of "how long" is probably worth a larger discussion.

Reply:
Yes. I don't know the answer. In this case, I guess my answer for this question is unless we have to change we don't change. I guess I'll remove this comment because we shouldn't do that in `DET-3262`